src/lib/realmdb/feedSource/NewsSource.js

```diff
@@ -0,0 +1,72 @@
+import moment from 'moment'
+import CeramicFeed from '../../ceramic/CeramicFeed'
+import FeedSource from './FeedSource'
+
+export default class NewsSource extends FeedSource {
+  // format from ceramic format to TreadDB format
+  static formatCeramicPost(ceramicPost) {
+    return {
+      _id: ceramicPost.id,
+      id: ceramicPost.id,
+      createdDate: moment(ceramicPost.published).format(),
+      date: moment(ceramicPost.published).format(),
+      displayType: 'news',
+      status: ceramicPost.hidden ? 'deleted' : 'published',
+      type: 'news',
+      data: {
+        reason: ceramicPost.content,
+        counterPartyFullName: ceramicPost.title,
+        subtitle: ceramicPost.title,
+        picture: ceramicPost.picture,
+        readMore: true,
+        link: ceramicPost.link,
+        sponsoredLink: ceramicPost.sponsored_link,
+        sponsoredLogo: ceramicPost.sponsored_logo,
+      },
+    }
+  }
+
+  async syncFromRemote() {
+    const { log, Feed, storage } = this
+    const ceramicCachedHistoryId = await storage.getItem('GD_CERAMIC_HISTORY')
+    log.info('ceramic sync from remote started', { ceramicCachedHistoryId })
+
+    if (!ceramicCachedHistoryId) {
+      const ceramicPosts = await CeramicFeed.getPosts()
+      log.debug('Ceramic fetched posts', ceramicPosts)
+
+      const formatedCeramicPosts = ceramicPosts.map(NewsSource.formatCeramicPost)
+
+      await Feed.save(...formatedCeramicPosts)
+
+      const historyId = await CeramicFeed.getHistoryId()
+      await storage.setItem('GD_CERAMIC_HISTORY', historyId)
+    } else {
+      const { history, historyId } = await CeramicFeed.getHistory(ceramicCachedHistoryId)
+      log.debug('Ceramic history', { history, historyId })
+
+      await Promise.all(
+        history.map(async ({ item: postId, action }) => {
+          log.debug('ceramic feed item action', { postId, action })
+
+          switch (action) {
+            case 'added':
+            case 'updated': {
+              const post = await CeramicFeed.getPost(postId)
+              await Feed.save(NewsSource.formatCeramicPost(post))
+              break
+            }
+            default: {
```

Comment:
'delete' should be explicit, it is too dangerous to assume all other commands are delete

Reply:
@alt-a1r @roma-hladilka 

Pls do 1. an exact case for delete 2. Add log.warn('Invalid action Received', { action }) to the default case